plugins/rocksdb/src/main/java/tech/pegasys/pantheon/plugin/services/storage/rocksdb/RocksDBPlugin.java

```diff
@@ -0,0 +1,108 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.plugin.services.storage.rocksdb;
+
+import tech.pegasys.pantheon.plugin.PantheonContext;
+import tech.pegasys.pantheon.plugin.PantheonPlugin;
+import tech.pegasys.pantheon.plugin.services.PicoCLIOptions;
+import tech.pegasys.pantheon.plugin.services.StorageService;
+import tech.pegasys.pantheon.plugin.services.storage.SegmentIdentifier;
+import tech.pegasys.pantheon.plugin.services.storage.rocksdb.configuration.RocksDBCLIOptions;
+import tech.pegasys.pantheon.plugin.services.storage.rocksdb.configuration.RocksDBFactoryConfiguration;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.google.auto.service.AutoService;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@AutoService(PantheonPlugin.class)
+public class RocksDBPlugin implements PantheonPlugin {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String NAME = "rocksdb";
+
+  private final RocksDBCLIOptions options;
+  private PantheonContext context;
+  private RocksDBKeyValueStorageFactory factory;
+  private RocksDBKeyValuePrivacyStorageFactory privacyFactory;
+
+  public RocksDBPlugin() {
+    this.options = RocksDBCLIOptions.create();
+  }
+
+  @Override
+  public void register(final PantheonContext context) {
+    LOG.info("Registering plugin");
+    this.context = context;
+
+    context
+        .getService(PicoCLIOptions.class)
+        .ifPresentOrElse(
+            picoCLIOptions -> picoCLIOptions.addPicoCLIOptions(NAME, options),
+            () -> LOG.error("Failed to register CLI options due to missing PicoCLIO options."));
+
+    context
+        .getService(StorageService.class)
+        .ifPresentOrElse(
+            this::createAndRegister,
+            () -> LOG.error("Failed to register KeyValueFactory due to missing StorageService."));
+
+    LOG.info("Plugin registered.");
+  }
+
+  @Override
+  public void start() {
+    LOG.info("Starting plugin.");
+    if (factory == null) {
+      LOG.debug("Applied configuration: {}", options.toString());
+      context
+          .getService(StorageService.class)
+          .ifPresentOrElse(
+              this::createAndRegister,
+              () -> LOG.error("Failed to register KeyValueFactory due to missing StorageService."));
+    }
+  }
+
+  @Override
+  public void stop() {
+    try {
+      if (factory != null) {
+        factory.close();
+        factory = null;
+      }
+
+      if (privacyFactory != null) {
+        privacyFactory.close();
+        privacyFactory = null;
+      }
+    } catch (final IOException e) {
+      LOG.error("Failed to stop plugin: {}", e.getMessage(), e);
```

Comment:
since we are only logging error, does it make sense to try/catch factory and privacyFactory separately?

Reply:
Probably, done.